core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -795,6 +796,13 @@ private RexNode convertCall(
         for (RexNode expr : oldExprs) {
           exprs.add(cx.getRexBuilder().ensureType(type, expr, true));
         }
+      } else if (!SqlTypeUtil.areSameFamily(RexUtil.types(exprs))
+          && consistency != SqlOperandTypeChecker.Consistency.NONE) {
```

Comment:
Sorry, i don't think we should make any decision and throw assertion error like this, the StandardConvertletTable.java did some work for the function call's operands type consistency, but it does not guarantee this, and it only touch the standard operator table, how about the UDF/UDAF ?

Also calcite allows some comparable types to be compared directly, like a .string and numeric, so please do not make the change to these tests, this destroy the semantics.

Reply:
Yes, ~ This PR is lack of consideration, I will close it and rethink about it
THX for comment